frontend/src/scenes/insights/InsightTabs/TrendTab/TrendTab.tsx

```diff
@@ -10,12 +10,59 @@ import { InfoCircleOutlined } from '@ant-design/icons'
 import { trendsLogic } from '../../trendsLogic'
 import { ViewType } from '../../insightLogic'
 import { LIFECYCLE } from 'lib/constants'
+import { featureFlagLogic } from 'lib/logic/featureFlagLogic'
 
-export function TrendTab(): JSX.Element {
-    const { filters } = useValues(trendsLogic({ dashboardItemId: null, view: ViewType.TRENDS }))
-    const { setFilters } = useActions(trendsLogic({ dashboardItemId: null, view: ViewType.TRENDS }))
+interface TrendTabProps {
+    view: string
+}
+
+export function TrendTab({ view }: TrendTabProps): JSX.Element {
+    const { filters } = useValues(trendsLogic({ dashboardItemId: null, view }))
+    const { setFilters } = useActions(trendsLogic({ dashboardItemId: null, view }))
+    const { featureFlags } = useValues(featureFlagLogic)
 
-    return (
+    return featureFlags['remove-shownas'] ? (
+        <>
+            <h4 className="secondary">{'Actions & Events'}</h4>
+            <ActionFilter
+                filters={filters}
+                setFilters={(payload): void => setFilters(payload)}
+                typeKey="trends"
+                hideMathSelector={filters.shown_as === LIFECYCLE}
+                copy="Add graph series"
+                disabled={filters.shown_as === LIFECYCLE && (filters.events?.length || filters.actions?.length)}
+            />
+            <hr />
+            <h4 className="secondary">Filters</h4>
+            <PropertyFilters pageKey="trends-filters" />
+            {filters.insight === ViewType.TRENDS && <hr />}
```

Comment:
Nit: Seems like this and the next 3 conditional elements can be under the same `if` conditional just under a single React segment.

Reply:
yup